### PR TITLE
doc: fix spec typos

### PR DESCRIPTION
**Pull request by 0xYYY**
**Description**

Fix some typos in the spec and `CONTRIBUTING.md`.

### CODE REVIEW

**Review by smartcontracts — APPROVED** [review]

**Comment by smartcontracts** [comment]
Thank you!

File: specs/introduction.md [hunk 1]
```diff
@@ -60,13 +60,13 @@ In order to scale Ethereum without sacrificing security, we must preserve 3 crit
 liveness, availability, and validity.
 
 1. **Liveness** - Anyone must be able to extend the rollup chain by sending transactions at any time.
```
Comment (tynes) [inline specs/introduction.md:62]: Slight nit here: numbered lists in markdown should always use `1` and the markdown rendering client will automatically increment them, this is to reduce diffs when making changes

File: specs/rollup-node.md [hunk 2]
```diff
@@ -72,4 +72,4 @@ The input and return types here are as defined by the [engine API specs][engine-
         OR `String` - one of `"safe"`, `"latest"`, or `"pending"`.
 - returns:
   1. `version`: `DATA`, 32 Bytes - the output root version number, beginning with 0.
-  1. `l2OutputRoot`: `DATA`, 32 Bytes - the output root
+  2. `l2OutputRoot`: `DATA`, 32 Bytes - the output root.
```
Comment (tynes) [inline specs/rollup-node.md:75]: Can you undo this diff please

Comment (0xYYY) [reply specs/rollup-node.md:75]: sure, i'll revert this

File: specs/introduction.md (hunk 1 above)
Comment (0xYYY) [reply specs/introduction.md:62]: then i think we can use this opportunity to change all the numbers here to `1`?

**Review by smartcontracts — APPROVED** [review]

**Comment by tynes** [comment]
Thank you!

**Pull request merged**